### PR TITLE
New triangle collapse arrow for large collapsible header

### PR DESCRIPTION
**Pull request by abey79**
### What

This PR implements the proper Figma-based collapsing arrow (with "linearised" rounded corners) and uses it for the large collapsible headers in the selection panel. This PR also fixes the vertical optical alignment of the text within these headers.

Partially addresses #2738 

Zoom 150%:
image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/2920) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/2920)
- [Docs preview](https://rerun.io/preview/pr%3Aantoine%2Fcollapsing-triangle/docs)
- [Examples preview](https://rerun.io/preview/pr%3Aantoine%2Fcollapsing-triangle/examples)

### CODE REVIEW

**Review by martenbjork — APPROVED** [review]
Looks good visually!

As we discussed on the huddle, there will be many icons, so it's good if adding icons isn't messy code wise or takes a lot of time. Some approaches we talked about:

- Linearizing the paths like this PR does
- Extracting path data from SVG attributes
- Apple approach: Adding multiple images like icon@2x.png, icon@3x.png etc and choosing them based on ui zoom level.

Anyways, I'll leave this to you!

**Comment by abey79** [comment]
Yes, that's on my long list of topics to run by @emilk.

That being said and regardless of where we're going with icons in general, the collapsing arrow may deserve special treatment because of the rotation animation. I think this is the lens with which this PR should be evaluated.

**Comment by abey79** [comment]
One clear shortcoming of this PR is that it doesn't document how I came up with the coordinates. I don't think the process is clean enough to be included though. For reference, here is the command I used:
```
vpype read -q 0.15 arrow.svg scale -o 0 0 %100/12% %100/12% pagesize 100x100 scale -o 50 50 %75/50% jinja -p rustvec.txt
```
where:
- arrow.svg is a SVG I manually exported from Figma
- `jinja -p rustvec.txt` is a command from a new, yet unpublished vpype plugin of mine 😓

**Comment by abey79** [comment]
Icon discussion moved here: https://github.com/rerun-io/rerun/issues/2960

File: crates/re_ui/src/lib.rs [hunk 1]
```diff
@@ -531,6 +536,44 @@ impl ReUi {
...
+            pos2(0.316248, 1.017085),
+            pos2(0.286141, 1.029362),
+            pos2(0.257726, 1.017592),
+            pos2(0.245118, 0.987622),
+            pos2(0.245118, 0.012378),
+            pos2(0.257726, -0.017592),
+            pos2(0.286141, -0.029362),
+            pos2(0.316248, -0.017085),
+            pos2(0.80387, 0.470537),
+        ];
```
Comment (emilk) [inline crates/re_ui/src/lib.rs:562]: Perhaps link to https://github.com/rerun-io/rerun/issues/2960 here so we remember to update this code once that issue is resolved

**Review by emilk — APPROVED** [review]
Looks good!

**Pull request merged**